aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -167,12 +166,18 @@ module aptos_framework::fungible_asset {
         );
         move_to(metadata_object_signer,
             Metadata {
-                supply,
                 name,
                 symbol,
                 decimals,
             }
         );
+        if (option::is_some(&monitoring_supply_with_maximum)) {
+            let maximum = option::destroy_some(monitoring_supply_with_maximum);
+            move_to(metadata_object_signer, Supply {
+                current: 0,
+                maximum
+            });
+        };
```

Comment:
do we offer optional supply in coin? if so, fine, if not, not so great....

Reply:
yes, this is why I follow the pattern.
FYI: https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/coin.move#L462

Reply:
We can actually considering always having a supply if we can figure out how to help existing fungible assets migrate to aggregator later on when aggregator is read